### DISCUSSION TITLE
Using presidio with dataframe

### DISCUSSION THREAD

**Discussion by Onenguyen** [microsoft/presidio#745] [Q&A] [answered]
I'm trying to use Presidio with a dataframe and I'm getting an error:

> TypeError: Argument 'string' has incorrect type (expected str, got DataFrame)

Here's what I have so far:

```
import pandas as pd
from presidio_analyzer import AnalyzerEngine

extractedtext = pd.read_csv('data/BuddyGreg_1119_Test_utf8.csv', encoding='utf-8')

analyzer = AnalyzerEngine()

results = analyzer.analyze(text=extractedtext,
                           entities=["US_SSN"],
                           language='en',
                           return_decision_process=True)
print(results)
```
I'm really new to python and appreciate any help. Thank you!

**Comment by Onenguyen** [comment, marked as answer]
Just converted to str(extractedtext) and everything worked

**Reply by omri374** [reply]
Hi @Onenguyen, while this works, it treats the entire data frame as text. This could be an issue if Presidio detects an entity in your column name, for example. [This example notebook](https://github.com/microsoft/presidio/blob/main/docs/samples/python/batch_processing.ipynb) breaks the data frame into values, and performs analysis and anonymization on each value.
